lua/bufferline/layout.lua

```diff
@@ -45,43 +69,42 @@ end
 local function calculate(state)
   local opts = vim.g.bufferline
 
-  -- separator + icon + space-after-icon + space-after-name
+  local has_icons = (opts.icons == true) or (opts.icons == 'both')
+
+  -- separator [+ icon + space-after-icon] + name + space-after-name
   local base_width =
-      1 -- sign
-    + (opts.icons and 2 or 0)
+    (has_icons and (1 + 1) or 0) -- icon + space-after-icon
         -- name
-    + 1 -- space after name
-    + (opts.closable and 2 or 0)
+    + 1 -- space-after-name
 
```

Comment:
Why has most of the `base_width` logic been moved inside `calculate_buffers_width`? `base_width` should be the width of all elements of non-variable width, that means everything except for the buffer name, and the buffer index.

The only reason for doing so would be to support different-width separator active/inactive icons or different-width close-tab/close-tab-modified icons, but I don't have any intention to do that because first I don't see any use case for it, second no one has requested it.

Turning `(opts.closable and 2 or 0)` into 7 lines of code has a cost in terms of readability and maintainability and that cost needs to be justified. (same for `(opts.icons and 2 or 0)`)

Also `state.is_picking_buffer` shouldn't be used. `BufferPick` should not affect the layout of the bufferline, that would be bad UX because a layout shift means the user is disoriented. The letter of jump-mode should take the place of something already rendered in the bufferline to avoid this.